src/internal/fetch/types.ts

```diff
@@ -0,0 +1,25 @@
+import type { Headers } from 'cross-fetch';
+
+/**
+ * Interface for additional request options to be compatible with `node-fetch`.
+ *
+ * @remarks
+ * To keep the http service compatible with a potential node implementation using `node-fetch`,
+ * we want to keep the request options at a subset of the supported props both for the native
+ * and the node-based fetch api.
+ */
+export interface RequestOptions {
+    body?: string | null;
+    headers?: string[][] | Record<string, string> | Headers;
+    method?: string;
+    redirect?: 'error' | 'follow' | 'manual';
+    signal?: AbortSignal | null;
+}
+
+export type GetPayload = string | string[][] | Record<string, string | number | boolean> | URLSearchParams;
+
+export type PostPayload = unknown;
+
+export type RequestPayload = GetPayload | PostPayload;
+
+export type RequestMethod = 'GET' | 'POST';
```

Comment:
while i bet there are various libraries to "normalize" fetch x-platform, i'm pretty damn sure I'd prefer a lean hand-roll to some _moar npm bloat_ 

:+1:

Reply:
under the hood i'm actually using the `cross-fetch` package, but it's pretty lean. when used in browsers, it simply returns the native browser fetch API, when on node, it uses a w3c compliant fetch implementation (`node-fetch`).
i just added some additional typings around it, to ensure max type compatibility between browser fetch and `node-fetch`, as they have a few differences (which are documented and understandable...).